lib/iris/tests/test_analysis.py

```diff
@@ -365,6 +365,26 @@ def test_percentile_1d(self):
                        ('analysis', 'third_quartile_foo_1d.cml'),
                        checksum=False)
 
+    def test_fast_percentile_1d(self):
+        cube = tests.stock.simple_1d()
+
+        first_quartile = cube.collapsed('foo', iris.analysis.PERCENTILE,
+                                        percent=25, percentile_method='fast')
+        np.testing.assert_array_almost_equal(first_quartile.data,
+                                             np.array([2.5], dtype=np.float32))
+        self.assertCML(first_quartile, ('analysis',
```

Comment:
I have rewritten the unit tests across the section of interest. I have (possibly unwisely) removed the contentious assertCML tests. Instead I have added a simple additional coordinate check on the resulting cubes as something approaching a replacement, though obviously not as rigorous and it could be expanded. I have moved the assert statements into functions and separated out the cases where the first and third percentiles were being checked in a single test.